templates/chat/src/components/ChatMessage.tsx

```diff
@@ -0,0 +1,61 @@
+import { type UIMessage } from '@ai-sdk/react'
+import ReactMarkdown from 'react-markdown'
+import { TldrawProviderMetadata } from './WhiteboardModal'
+
+interface ChatMessageProps {
+	message: UIMessage
+	onImageClick: (tldrawMetadata: TldrawProviderMetadata) => void
+}
+
+export function ChatMessage({ message, onImageClick }: ChatMessageProps) {
+	// Split the message parts into image and text ones
+	const textParts = message.parts.filter((part) => part.type === 'text' && part.text.trim())
+	const imageParts = message.parts.filter((part) => part.type === 'file')
+
+	// For AI messages with no content, show thinking state
+	if (message.role === 'assistant' && textParts.length === 0 && imageParts.length === 0) {
+		return (
+			<div className="message assistant-message thinking-message">
+				<div className="thinking-text">Thinking…</div>
```

Comment:
totally unnecessary but if you wanted to do a nice animation with the ellipsis might be a nice "loading" touch